src/types/presentation_time.zig

```diff
@@ -0,0 +1,66 @@
+const wlr = @import("wlroots.zig");
+
+const wayland = @import("wayland");
+const wl = wayland.server.wl;
+
+const Presentation = extern struct {
+    pub const Event = extern struct {
+        output: ?*wlr.Output,
+        tv_sec: u64,
+        tv_nsec: u32,
+        refresh: u32,
+        seq: u64,
+        flags: u32,
+    };
+
+    global: *wl.Global,
+    feedbacks: wl.list.Head(PresentationFeedback, "link"),
+    clock: c_int,
+
+    events: extern struct {
+        destroy: wl.Signal(*wlr.Presentation),
+    },
+
+    server_destroy: wl.Listener(*wl.Server),
+
+    extern fn wlr_presentation_create(server: *wl.Server, backend: *wlr.Backend) ?*wlr.Presentation;
+    pub fn create(server: *wl.Server) !*wlr.Presentation {
+        return wlr_presentation_create(server) orelse error.OutOfMemory;
+    }
+
+    extern fn wlr_presentation_surface_sampled(presentation: *wlr.Presentation, surface: *wlr.Surface) *wlr.PresentationFeedback;
```

Comment:
This can return null.
```suggestion
    extern fn wlr_presentation_surface_sampled(presentation: *wlr.Presentation, surface: *wlr.Surface) ?*wlr.PresentationFeedback;
```